File: sieve_of_eratosthenes/SieveOfEratosthenes.rb
require 'time'

class SieveOfEratosthenes
  arr = Array.new(100, 1)

  i = 0
  k = 2
  size = arr.size

  start_time = Time.now

  puts "=== Find Prime Numbers in Ruby ==="

  while (k * k) <= size
    i = k

    while (i * k) < size
      arr[k * i] = 0
      i += 1
    end

    begin
      k += 1
    end while arr[k] == 0
  end

  i = 2

  while i < size
    puts i if arr[i] == 1
    i += 1
  end

  end_time = Time.now

  puts "Exec time: #{(end_time - start_time) * 100000} ms"
end

File: sieve_of_eratosthenes/SieveOfEratosthenes.java
import java.util.*;

/**
 * SieveOfEratosthenes
 */
public class SieveOfEratosthenes {

  public static void main(String[] args) {
    int arr[] = new int[100];
    Arrays.fill(arr, 1);

    int size = arr.length;
    int i = 0;
    int k = 2;

    long start = System.nanoTime();

    System.out.println("=== Find Prime Numbers in Java ===");

    while(Math.pow(k, 2) <= size) {
      i = k;

      while(i * k < size) {
        arr[k * i] = 0;
        i++;
      }

      do {
        k++;
      } while (arr[k] == 0);
    }

    i = 2;

    while(i < size) {
      if(arr[i] == 1) {
        System.out.println(i);
      }
      i++;
    }

    long end = System.nanoTime();

    System.out.println("Exec time: " + (end - start) / 1000000f + " ms");
  }
}